python/mlx/utils.py

```diff
@@ -191,3 +191,45 @@ def tree_unflatten(tree):
         return l
     else:
         return {k: tree_unflatten(v) for k, v in children.items()}
+
+
+def tree_reduce(fn, tree, initializer=None, is_leaf=None):
+    """
+    Recursively applies a reducer function `fn` to the leaves of the Python tree `tree` and aggregates them into a single value.
```

Comment:
The word "recursive" here is slightly confusing. I would shorten this. There's enough detail on usage below.

```suggestion
    Applies a reduction to the leaves of a Python tree.
```